apollo-router/src/plugins/rhai.rs

```diff
@@ -1,105 +1,249 @@
+                ) -> Result<Context, Box<EvalAltResult>> {
+                    obj.with_mut(get_context)
+                }
+
+                pub fn [<get_context_ $base _response>](
+                    obj: &mut [<Shared $base:camel Response>],
+                ) -> Result<Context, Box<EvalAltResult>> {
+                    obj.with_mut(get_context)
+                }
+
+                pub fn [<insert_context_ $base _request>](
+                    obj: &mut [<Shared $base:camel Request>],
+                    context: Context
+                ) -> Result<(), Box<EvalAltResult>> {
+                    obj.with_mut(|request| insert_context(request, context))
+                }
+
+                pub fn [<insert_context_ $base _response>](
+                    obj: &mut [<Shared $base:camel Response>],
+                    context: Context
+                ) -> Result<(), Box<EvalAltResult>> {
+                    obj.with_mut(|response| insert_context(response, context))
+                }
+
+                pub fn [<get_originating_headers_ $base _request>](
+                    obj: &mut [<Shared $base:camel Request>],
+                ) -> Result<HeaderMap, Box<EvalAltResult>> {
+                    obj.with_mut(get_originating_headers)
+                }
+
+                pub fn [<get_originating_body_ $base _request>](
+                    obj: &mut [<Shared $base:camel Request>],
+                ) -> Result<Request, Box<EvalAltResult>> {
+                    obj.with_mut(get_originating_body)
+                }
+
+                /* XXX ONLY VALID FOR CERTAIN TYPES
+                 *
+                pub fn [<get_originating_headers_ $base _response>](
+                    obj: &mut [<Shared $base:camel Response>],
+                ) -> Result<HeaderMap, Box<EvalAltResult>> {
+                    obj.with_mut(get_originating_headers)
+                }
+                */
+
+                pub fn [<set_originating_headers_ $base _request>](
+                    obj: &mut [<Shared $base:camel Request>],
+                    headers: HeaderMap
+                ) -> Result<(), Box<EvalAltResult>> {
+                    obj.with_mut(|request| set_originating_headers(request, headers))
+                }
+
+                pub fn [<set_originating_body_ $base _request>](
+                    obj: &mut [<Shared $base:camel Request>],
+                    body: Request
+                ) -> Result<(), Box<EvalAltResult>> {
+                    obj.with_mut(|request| set_originating_body(request, body))
+                }
+
+                /* XXX ONLY VALID FOR CERTAIN TYPES
+                pub fn [<set_originating_headers_ $base _response>](
+                    obj: &mut [<Shared $base:camel Response>],
+                    headers: HeaderMap
+                ) -> Result<(), Box<EvalAltResult>> {
+                    obj.with_mut(|response| set_originating_headers(response, context))
+                }
+                */
```

Comment:
is it still useful ?

Reply:
I'll leave a comment that explains more clearly why this is excluded from the macro and remove the code block.

Reply:
done: c80221d1eb58838468c0813377efce814008b502